src/rules/at-import-no-partial-extension/index.js

```diff
@@ -0,0 +1,65 @@
+import { isRegExp, isString } from "lodash"
+import { utils } from "stylelint"
+import nodeJsPath from "path"
+
+export const ruleName = "at-import-no-partial-extension"
+
+export const messages = utils.ruleMessages(ruleName, {
+  expected: "Unexpected file extension in imported partial name",
+})
+
+export default function (on, options) {
+  return (root, result) => {
+    const validOptions = utils.validateOptions(result, ruleName, {
+      actual: on,
+    }, {
+      actual: options,
+      possible: {
+        // Accepting array of either strings or regular expressions
+        ignoreExtensions: str => isRegExp(str) || isString(str),
+      },
+      optional: true,
+    })
+    if (!validOptions) { return }
+
+    function checkPathForUnderscore(path, decl) {
+      // Stripping trailing quotes and whitespaces, if any
+      const pathStripped = path.replace(/^\s*?("|')\s*?/, "").replace(/\s*?("|')\s*?$/, "")
```

Comment:
I think that there should be some tests that include whitespace that these regexes should match. The tests are passing if I leave only the part that replaces quotes:

```js
const pathStripped = path.replace(/("|')$/, "")
```